### ISSUE TITLE
[bug] NormalizeReward is wrong?

### ISSUE THREAD

**Issue by colinrgodsey** [openai/gym#3289]
**Describe the bug**
I think `NormalizeReward` is wrong in two places... I could be missing something, but... it does not seem right. I'm going to feel really stupid if this is right lol

With that said, I don't know if this could be fixed without just breaking reproducibility. 

**Code example**
This isn't a gamma running average:

https://github.com/openai/gym/blob/dcd185843a62953e27c2d54dc8c2d647d604b635/gym/wrappers/normalize.py#L133

It should be:
```python
self.returns = self.returns *  (1.0 - self.gamma) + rews * self.gamma
```

I feel like this should just explode in value.

It's also not using the mean when it actually normalizes:

https://github.com/openai/gym/blob/dcd185843a62953e27c2d54dc8c2d647d604b635/gym/wrappers/normalize.py#L144

Should be:
```python
return (rews -  self.return_rms.mean) / np.sqrt(self.return_rms.var + self.epsilon)
```

**System Info**

**Additional context**

### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/openai/gym/issues) in the repo (**required**)

**Comment by colinrgodsey** [comment]
And I do see it's this way in stable baselines too, which is where this code came from, but....

**Comment by colinrgodsey** [comment]
nvm.... this seems to be a long string of discussion over at SB3....

https://github.com/DLR-RM/stable-baselines3/issues/348#issuecomment-795147337

**Comment by colinrgodsey** [comment]
well, i'll let someone else close this out if im wrong, but:

it seems in SB3, that set of math is used to normalize *returns*, not rewards, and the gamma used is expected to match the GAE (or normal?) gamma.

**Comment by pseudo-rnd-thoughts** [comment]
The rewards are normalised by the running returns, not by the running rewards as you note
It doesn't make much sense to me but work has found this to be more helpful for some reason.

**Comment by colinrgodsey** [comment]
I have also used this normalization fine in my PPO work, and it's been fine. I adjusted my reward normalization using the "correct" normalization and now i get exploding gradients in my value branch, which leads to it producing a policy that deliberately tries to lower the reward (objectively and consistently worse than the random policy) :sob: 

Oddly, if I pretrain just the value branch for enough time for the value branch to learn the scale, then I can train like normal. I'm thinking using the "incorrect" reward normalization just makes the values very very low when going into the value loss which just stabilizes things

**Comment by colinrgodsey** [comment]
“Implementation matters in deep RL”, by Engstorm et al.   Apendex A.2, there it is...
